app/controllers/brands_controller.rb

```diff
@@ -0,0 +1,32 @@
+class BrandsController < ApplicationController
+  def index
+    @brands = Brand.all
+  end
+
+  def create
+    @brand = Brand.new(name: params[:name])
+    if @brand.save
+      redirect_to brands_path, notice: "New brand created successfully"
+    else
+      redirect_to brands_path, alert: @brand.errors.full_messages
+    end
+  end
+
+  def update
+    @brand = Brand.find(params[:id])
+    if @brand.update(name: params[:new_name])
+      redirect_to brands_path, notice: "Brand name updated successfully"
+    else
+      redirect_to brands_path, alert: @brand.errors.full_messages
+    end
+  end
+
+  def destroy
+    @brand = Brand.find(params[:id])
```

Comment:
line 16, 25 are the same. Rails have "Don't repeat your self" principle 
We can deal with callback "before_action". 
Like this https://github.com/thanhnguyen2612/phone_shop/blob/develop/app/controllers/inventories_controller.rb#L41-L43

Reply:
ok I'll practice this